.github/workflows/ci.yml

```diff
@@ -121,6 +123,10 @@ jobs:
       uses: ./smithy-rs/.github/actions/docker-build
       with:
         action: ${{ matrix.test.action }}
+    - name: Install mdbook
+      shell: bash
+      run: |
+        cargo install mdbook
```

Comment:
Can this get its own CI job separate from the Docker image, similar to the semver checks job? Or even better, give it its own workflow file so that you can make it only run when the `design/` directory is changed.

Putting it in normal CI will slow all PR CI runs down for something that rarely changes.

Reply:
> only run when the `design/` directory is changed

I don't think this is sufficient. If the underlying API changes the `mdbook test` might break.

Reply:
Then `mdbook` needs to be added to the `Dockerfile` so that we're not installing it every test run.